### ISSUE TITLE
Validate label groups/prefixes

### ISSUE THREAD

**Issue by devnev** [agilepathway/label-checker#328]
# Problem

I'd like to be able to write validation rules for certain label groups/prefixes, e.g. `type: fix` or `visibility/public` - something like "at most one from this group", "exactly one from this group", and maybe "one of a subset from this group" or even "none from this group".

Looking around existing github repositories, group prefixes could be delimited with `:` or `/`, so that could maybe be kept flexible.

# Possible solutions

A possible approach might be to extend the specification with a glob or regex syntax, and expand the list matching labels, then apply the rule - so e.g.

```
one_of: "type: *"
```

would expand the list to all the matching `type: `-prefixed labels on the PR (or on the repo?), then validate that there's only one. this works with the other options too

```
one_of: "type: *" # exactly one matching
none_of: "type: *" # none matching
any_of: "type: *" # at least one matching
all_of: "type: *" # this one doesn't make much sense if it expands to the labels on the PR, but makes sense if it expands to all matching labels of the repository
```

**Comment by johnboyes** [comment]
Hi @devnev, thanks for this feature suggestion.  How about the following, as a potential implementation?

- introduce a new optional [GitHub Actions input parameter](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs] named something like `check-prefix-only`, which would be a boolean and default to `false`.

e.g. a couple of examples from your scenarios above:

```
        with:
          check-prefix-only: true
          one_of: "type:"
          repo_token: ${{ secrets.GITHUB_TOKEN }}
```

```
        with:
          check-prefix-only: true
          none_of: "visibility/"
          repo_token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by devnev** [comment]
Thanks @johnboyes, yes I think that would work too.

Another benefit of having this prefix-based matching would be to eliminate redundancy of both the setup of the labels and the label checker configuration having to enumerate a set of labels.

**Comment by johnboyes** [comment]
Great, I'll try to get round to implementing this soon.  My reasoning for having a dedicated `check-prefix-only` input param is that it's a bit more declarative and intentful I reckon, less opportunity for subtle bugs.  Sound sensible? Having a regex style syntax would imply that other regexes as well as `*` could be used, which then makes me think of this: 
https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/ 😉

**Comment by devnev** [comment]
100% agreed.

**Comment by johnboyes** [comment]
@devnev just to let you know I've started work on this, and it is proving straightforward to implement.  Aiming to have it released in the next week.

**Comment by johnboyes** [comment]
Hi @devnev, I've released this in [v1.4.0](https://github.com/agilepathway/label-checker/releases/tag/v1.4.0).  Also see [the new documentation for this feature](https://github.com/agilepathway/label-checker#match-labels-based-on-prefix).  Thanks for the suggestion, and any feedback on whether it fits your needs is really appreciated 🙏🏻

**Comment by devnev** [comment]
Thanks @johnboyes, its working as expected
